lib/THC/generic/THCTensorMath.cu

```diff
@@ -391,4 +391,42 @@ accreal THCTensor_(trace)(THCState *state, THCTensor *src_) {
   return trace;
 }
 
+#if defined(THC_REAL_IS_FLOAT) || defined(THC_REAL_IS_DOUBLE) || defined(THC_REAL_IS_HALF)
+
+void THCTensor_(linspace)(THCState *state, THCTensor *r_, real a, real b, long n) {
+  THCAssertSameGPU(THCTensor_(checkGPU)(state, 1, r_));
```

Comment:
do what jeff suggests, i.e.: make the kernel for contiguous tensors, and then you can do a freeCopyTo to copy it back to the result tensor.
You can search the repo for freeCopyTo to see this pattern